_ts/widget.ts

```diff
@@ -0,0 +1,137 @@
+
+        var widgets: NodeListOf<HTMLElement> = document.querySelectorAll('.github-widget');
+        for(var i:number = 0; i < widgets.length; i ++)
+        {
+            var parentNode: HTMLElement = widgets[i];
+            var type: string = parentNode.dataset.type;
+            var widget_name: string = type + "_widget" + i;
+            var keyword:string = "";
+
+            if(type === "search")
+            {
+                keyword = (<HTMLInputElement>document.getElementById('keyword')).value;
+            }
+            else
+            {
+                keyword = parentNode.dataset.browsetopic;
+            }
+
+            parentNode.setAttribute("id", widget_name);
+            this.appendToWidget("#" + widget_name, "div", "", '<div class="gh-widget-container"><div class="gh-widget-item gh-widget-photo"></div><div class="gh-widget-item gh-widget-personal-details"></div></div><div class="gh-widget-container gh-widget-stats"></div><hr class="gh-widget-hr"><div class="gh-widget-container"><div class="gh-widget-item gh-widget-heading">Top repositories for "' + keyword + '"</div></div><div class="gh-widget-repositories"></div><div class="gh-widget-container"><div class="gh-widget-item gh-widget-follow"></div><div class="gh-widget-item gh-widget-active-time"></div></div>')
+            this.fetchRepos(keyword, "#"+widget_name);
+        }
+    }
+
+    fetchRepos(keyword: string, widgetId: string){
+        var repos: Array<any> = [];
+        for(var i: number = 0; i < this.accounts.length; i++)
+        {
+            var url:string = "https://api.github.com/search/repositories?q=topic:" + keyword + "&sort=stars&per_page=100/repos/"+this.accounts[i] + "/";
+            repos.push(this.getJSON(url));
+        }
+
+        this.updateRepoDetails(this.topRepos(repos), widgetId);
+        this.updateLastPush(this.lastPushedDay(repos), widgetId);
+
+        
+    }
+
+    appendToWidget(parentSelector:string, tag:string, classes:string, html:string){
+        var parentNode: HTMLElement = document.querySelector(parentSelector);
+        var childNode: HTMLElement = document.createElement(tag);
+        childNode.innerHTML = html;
+        childNode.className += classes;
+        parentNode.appendChild(childNode);
+    }
+
+    getJSON(url: string){
+        var request: XMLHttpRequest = new XMLHttpRequest();
+        request.open('GET', url, true);
+        request.setRequestHeader('Accept', 'application/vnd.github.mercy-preview+json');
+        request.onload = function() {
+            if (request.status === 200) {
+                var data = JSON.parse(request.responseText);
+                return data;
+            }
+        };
+        request.send();
```

Comment:
Couple of things here (not necessarily need to be in this PR): 
1. github API has throttling, https://developer.github.com/v3/rate_limit/. How can we handle the throttling? For example, for browse, we may have cached data to display.
2. let's think a way to measure the latency of this API and collect the data. This is something can impact the UX.